### DISCUSSION TITLE
Span ID is empty in java application

### DISCUSSION THREAD

**Discussion by raghu-megh** [open-telemetry/opentelemetry-java-instrumentation#2435] [Q&A] [answered]
Hello,

I am using the sampleapp (java) with auto instrumentation java agent jar. When I access the span ID from within the spring boot app I get this 0000000000000000 and trace ID is also only zeroes. Am I missing anything? However, I see the trace in my aws-xray console.

I am running the aws-otel-collector as docker.

Thanks,
Raghuram

**Comment by trask** [comment, marked as answer]
hi @raghu-megh! what version of the opentelemetry-javaagent are you using, and what version of the opentelemetry-api are you using? it sounds like you may be using incompatible versions

**Reply by raghu-megh** [reply]
Hi @trask 

I am using opentelemetry-javaagent - version: 0.15.0-aws.1 and api version 1.0

**Reply by raghu-megh** [reply]
I used 0.15 for api and java agent and it worked. Thanks @trask

**Comment by kriskrishna** [comment]
implementation(platform("io.opentelemetry:opentelemetry-bom:0.17.1"))
    implementation 'io.opentelemetry:opentelemetry-extension-annotations:0.16.0'
    implementation 'io.opentelemetry:opentelemetry-api:0.16.0'
    implementation 'io.opentelemetry:opentelemetry-context:1.13.0'
    implementation "io.opentelemetry.javaagent:opentelemetry-javaagent:1.11.1"
    implementation "io.opentelemetry:opentelemetry-sdk:1.13.0"
    implementation "io.opentelemetry:opentelemetry-exporter-otlp:1.13.0"

I am using above in spring-boot. And , getting  this spanId 0000000000000000 and trace ID is also only zeroes. Pls help.

**Reply by trask** [reply]
hi @kriskrishna! can you open a new discussion and provide a lot more details, or even better, a repro?
